metrics/web/lib/common/presentation/navigation/state/navigation_notifier.dart

```diff
@@ -106,20 +106,25 @@ class NavigationNotifier extends ChangeNotifier {
     if (_isAppInitialized) _redirect();
   }
 
+  /// Determines whether the current page can be popped.
+  bool canPop() {
+    return _pages.length > 1;
+  }
+
   /// Removes the current page and navigates to the previous one.
   void pop() {
-    if (_pages.length <= 1) return;
-
-    _pages.removeLast();
+    if (canPop()) {
```

Comment:
📱 
Not sure the `.pop` method itself should rely on the `canPop` method. Instead, every action that wants to pop a page should decide whether it relies on the canPop or not.